### PR TITLE
Make LocalTrajectoryBuilder optional.

### PR DESCRIPTION
**Pull request by cschuet**
This is to prepare for the cloud-based mapping case where the robot is running cartographer in pure-localization mode and the cloud instance only solves the global problem. In that case a `LocalTrajectoryBuilder` need to be instantiated in for the cloud instance.

### CODE REVIEW

File: cartographer/internal/mapping/global_trajectory_builder.h [hunk 1]
```diff
@@ -44,8 +44,10 @@ class GlobalTrajectoryBuilder : public mapping::TrajectoryBuilderInterface {
   void AddSensorData(
       const std::string& sensor_id,
       const sensor::TimedPointCloudData& timed_point_cloud_data) override {
+    CHECK(local_trajectory_builder_)
+        << "Cannot add TimedPointCloudData without a LocalTrajectoryBuilder";
```
Comment (wohe) [inline cartographer/internal/mapping/global_trajectory_builder.h]: nit: End sentence with a period.

File: cartographer/internal/mapping/global_trajectory_builder.h [hunk 2]
```diff
@@ -44,8 +44,10 @@ class GlobalTrajectoryBuilder : public mapping::TrajectoryBuilderInterface {
   void AddSensorData(
       const std::string& sensor_id,
       const sensor::TimedPointCloudData& timed_point_cloud_data) override {
+    CHECK(local_trajectory_builder_)
```
Comment (wohe) [inline cartographer/internal/mapping/global_trajectory_builder.h]: Add a comment about this in the header file, i.e. that `nullptr` is acceptable but no `TimedPointCloudData` can be added in this case?

File: cartographer/internal/mapping/global_trajectory_builder.h (hunk 1 above)
Comment (cschuet) [reply cartographer/internal/mapping/global_trajectory_builder.h]: Done.

File: cartographer/internal/mapping/global_trajectory_builder.h (hunk 2 above)
Comment (cschuet) [reply cartographer/internal/mapping/global_trajectory_builder.h]: Done.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**